examples/widgets/slider_demo.py

```diff
@@ -3,65 +3,82 @@
 Slider
 ======
 
-Using the slider widget to control visual properties of your plot.
+In this example, sliders are used to control the frequency and amplitude of
+a sine wave.
 
-In this example, a slider is used to choose the frequency of a sine
-wave. You can control many continuously-varying properties of your plot in
-this way.
+See :doc:`/gallery/widgets/slider_snap_demo` for an example of having
+the ``Slider`` snap to discrete values.
+
+See :doc:`/gallery/widgets/range_slider` for an example of using
+a ``RangeSlider`` to define a range of values.
 """
 import numpy as np
 import matplotlib.pyplot as plt
-from matplotlib.widgets import Slider, Button, RadioButtons
+from matplotlib.widgets import Slider, Button
+
+
+# The parametrized function to be plotted
+def f(t, amplitude, frequency):
+    return amplitude * np.sin(2 * np.pi * frequency * t)
 
-fig, ax = plt.subplots()
-plt.subplots_adjust(left=0.25, bottom=0.25)
 t = np.arange(0.0, 1.0, 0.001)
```

Comment:
```suggestion
t = np.arange(0, 1, 0.001)
```
While you're basically rewriting the whole example, let's clean up for Python 3 and reduce unnecessary clutter of `x.0` decimals.

Reply:
:+1: 

Also this is an odd usage of `arange` especially as it's docstring warns against using it for non-integer steps. I will also switch it to `linspace`